docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1036 @@
+In this case, the user argument for `workout /delete <workout index number in workout list>` is the workout index number of the workout to be deleted from the workout list.
+<br><br>
+**(Steps 7.3 to 7.6)** The `WorkoutList#deleteWorkout(userArgument)` method will then proceed to check whether the workout index number to be deleted
+is within the range of the workout list. This is done by calling the `WorkoutList#checkIndexIsWithinRange(indexToDelete)` method. If index to delete
+is within the range of the workout list, the method returns true, else it will return false.
+If the method returns false, then the `InvalidWorkoutException` exception is thrown.
+<br><br>
+**(Steps 7.7 to 7.8)** The `WorkoutList#deleteWorkout(userArgument)` method will then fetch the `Workout` object to be deleted
+by calling the `WorkoutList#getWorkoutFromIndexNum(indexToDelete)` method.
+<br><br>
+**(Steps 7.9 to 7.12)** The `Workout` object to be deleted is subsequently removed from the ArrayList and HashMap which stores the
+application's workout list.
+<br><br>
+**(Step 8)** The `WorkoutList#deleteWorkout(userArgument)` method returns the deleted `Workout` object to `WorkoutCommand`.
+<br><br>
+**(Steps 9 to 11)** Upon returning to the `WorkoutCommand` object, the `UI#printDeleteWorkoutMessage(deletedWorkout)` is called
+to display the workout that has been deleted to the user via the terminal. The following is an example 
+of a success deletion message after a valid workout is deleted from the application's workout list:
+```
+----------------------------------------------------------------------
+Alright, the following workout has been removed:
+
+	push up (20 reps)
+
+----------------------------------------------------------------------
+```
+**(Steps 12 to 13)** The `FileManager#rewriteAllWorkoutsToFile(workoutList)` is called to rewrite
+the `workouts.txt` file according to the newly modified application's workout list.
+<br><br>
+This completes the process of deleting an existing workout in WerkIt!
+
+##### Design Considerations for Deleting Existing Workout
+###### Rewrite All Workout To File
+Currently, the WerkIt! program will rewrite all workout to the resource file, `workouts.txt`, when the delete workout
+function is executed. Such implementation may have performance issues as the program needs to rewrite the whole
+file with the modified workout list whenever a workout is deleted in the application.
+<br><br>
+An alternative considered was to find the workout to be deleted in the resource file, and then
+remove that workout. While this is a more efficient implementation, it is more complex due to the
+way the workout data are formatted and stored in the `workouts.txt` file.
+<br><br>
+Hence, to simplify the implementation, the team decided to simply
+rewrite all workouts to the resource file whenever a workout is deleted.
+
+---
+
+#### Update Workout
+![Update Workout Class Diagram](uml/classDiagrams/images/updateWorkout.png)
```

Comment:
There's a bug with how this image is embedded - it's very blurry?
![image](https://user-images.githubusercontent.com/63784032/161183817-f1093d7a-c431-4d04-819b-5ce53bed6f88.png)